docker/compute_worker/compute_worker.py

```diff
@@ -607,7 +602,41 @@ def start(self):
             self.watch_file(),
             loop=loop,
         )
-        loop.run_until_complete(gathered_tasks)
+        signal.signal(signal.SIGALRM, alarm_handler)
+        signal.alarm(self.execution_time_limit)
+        try:
+            loop.run_until_complete(gathered_tasks)
+        except ExecutionTimeLimitExceeded:
+            pass
+        finally:
+            self.watch = False
+            for kind, logs in self.logs.items():
+                elapsed_time = logs["end"] - logs["start"] if logs["end"] is not None else self.execution_time_limit
```

Comment:
i don't think this needs to be an inline if, could be easier to understand broken out? what do you think?